privilege/privileges/tidb_auth_token.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package privileges
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"sync"
+	"time"
+
+	jwkRepo "github.com/lestrrat-go/jwx/v2/jwk"
+	jwsRepo "github.com/lestrrat-go/jwx/v2/jws"
+	jwtRepo "github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+// JWKSImpl contains a JSON Web Key Set (JWKS), and a filepath that stores the JWKS
+type JWKSImpl struct {
+	// contain one more JWKS for recover when failing to load the JWKS from the filepath.
+	cur, last jwkRepo.Set
```

Comment:
I prefer to only keep the effective JWKS in this structure and access it by `atomic` lib. It's clearer and faster.
E.g.
```
cur, err := jwkRepo.ReadFile(jwks.filepath)
if err == nil {
     atomic.Store(jwks.cur, cur)
}
```